### ISSUE TITLE
Equipping items in combat.

### ISSUE THREAD

**Issue by AshnakAGQ** [ZerdBG3/DnD-Rebalancing#20]
Right now you can equip/deequip items as you please in combat without using up an action. This can get really OP.

Using a greatsword? Get an extra d6 by equipping two light weapons and doing an offhand attack. Then switch to a shield before you end your turn for a free AC bonus.

**Comment by FelipeRenault** [comment]
I agree that is completely ridiculous. It's also strong for ranged weapons, since you should never pass your turn holding a bow. Always finish with a shield and sword so you get AC and AoO.
However, I think the action economy of switching gears is buried in a hardcoded mess. I'll check later if this is editable, but I'm afraid not. If it's not an easy fix, I'll wait to see if Larian will address it. If even then they don't fix it, I'll have to script in, which currently isn't possible without the engine.

**Comment by AshnakAGQ** [comment]
Equipping items took action points in Divinity II, so it's probably possible. I'm not sure how similar the two games' code are though.

**Comment by FelipeRenault** [comment]
There was a (hidden) talent that allowed the equipment switch to not cost action points, but the value itself wasn't moddable. And from what I've seen in BG3, it still isn't

**Comment by FelipeRenault** [comment]
Adding to this:
- Swapping melee for ranged setup should cost at least a bonus action
- While on one setup, you shouldn't have access to anything from the other setup (Eg.: AC from Shield while on your ranged weapon or a spell granted by one weapon while using the other)

**Comment by FelipeRenault** [comment]
As of Patch 7, switching equipment from the inventory in combat spends an action. You can still freely swap weapon sets, and it seems stats still apply regardless of current weapon set equipped (gain AC from shield while using ranged weapon, for example)
